plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchangeModule.java

```diff
@@ -68,6 +68,13 @@ else if (ImmutableSet.of("abfs", "abfss").contains(scheme)) {
             configBinder(binder).bindConfig(ExchangeAzureConfig.class);
         }
         else {
+            // Bind both S3 and ABFS config classes to handle the case when user made typo in the uri but provided extra
+            // filesystem-specific config options.
+            // If config classes  are not bound users will not observe the message from exception thrown below.
+            // Instead they will observe `Configuration property '...' was not used` error coming from configuration
```

Comment:
sounds like a problem in Airlift.
it's weird NOT to propagate Module's exception and instead tell the used about unused properties.
Properties being used or not is a result of running all the Modules, so upon a Module failure not so much can be said about unused properties.

Did you try to fix that there?

Reply:
I think it is a design choice. To validate config an fail fast before building injector at all. I think validation could be moved below Injector creation and it would also work. But startup would be slower in case of just config error.

Reply:
Makes sense. I think I am starting to understand.

However, the problem seems inherent to airlift configuration with conditional configuration options, right?
Like metastore type, or hive.security, etc. 

The difference here is that exchange plugin throws clearly in the Module class, while the above examples would result in a missing bind, but still the unused configuration would be reported.

- maybe exchange plugin shouldn't throw in the Module class? Maybe it can deal with the problem by validating its configuration that base directories' scheme is one of the supported ones
- maybe we should have an airlift change still, so that exception from Module setup is propagated (seems weird not to propagate it and throw some other exception instead)

cc @electrum